src/toxcore/state_format/old.rs

```diff
@@ -1055,38 +870,110 @@ impl SectionData {
         // TODO: don't return an empty Vec ?
         s.iter()
             .map(|sd| sd.as_section())
-            .filter(|s| s.is_some())
-            .map(|s| s.expect("IS Some(_)"))
+            .filter(|s| s.clone().is_done())
+            .map(|s| s.to_result().unwrap())
             .collect()
     }
 }
 
 
-from_bytes!(SectionData, do_parse!(
-    data_len: le_u32 >>
-    kind: call!(SectionKind::parse_bytes) >>
-    tag!(SECTION_MAGIC) >>
-    data: take!(data_len) >>
-    (SectionData { kind, data: data.to_vec() })
-));
+impl FromBytes for SectionData {
+    named!(from_bytes<SectionData>, do_parse!(
+        data_len: le_u32 >>
+        kind: call!(SectionKind::from_bytes) >>
+        tag!(SECTION_MAGIC) >>
+        data: take!(data_len) >>
+        (SectionData { kind, data: data.to_vec() })
+    ));
+}
+
+#[cfg(test)]
+impl Arbitrary for SectionKind {
+    fn arbitrary<G: Gen>(g: &mut G) -> Self {
+        let mut range_val: u8 = 7;
+
+        while range_val > 6 && range_val < 0x0a {
```

Comment:
It's just a way to generate random `SectionKind`. Generated value doesn't have to correspond to saved byte. You can assume that generated `7` means `SectionKind::TcpRelays` and `8` means `SectionKind::PathNodes` and remove this cycle.

Reply:
old.rs is deprecated.